proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -2521,6 +2439,445 @@ lemma possibleSwitchTo_all_invs_but_ct_not_inQ':
+          apply clarsimp
+          apply (rule corres_guard_imp)
+            apply (rule corres_split[OF _ gts_corres], rename_tac st st')
+              apply (rule_tac P="\<lambda>s. blocked_on_send_recv_tcb_at y s \<and> y \<noteq> idle_thread s
+                                     \<and> reply_unlink_ts_pred y s \<and> valid_sched s \<and> valid_tcbs s
+                                     \<and> pspace_aligned s \<and> pspace_distinct s
+                                     \<and> st_tcb_at ((=) st) y s"
+                          and P'="\<lambda>s. tcb_at' y s
+                                      \<and> st_tcb_at' (not runnable') y s
+                                      \<and> weak_sch_act_wf (ksSchedulerAction s) s
+                                      \<and> valid_queues s \<and> valid_queues' s \<and> valid_tcbs' s
+                                      \<and> valid_release_queue_iff s \<and> st_tcb_at' ((=) st') y s"
+                           in corres_inst)
+              apply (case_tac "\<exists>ep r_opt pl.
+                                st = Structures_A.thread_state.BlockedOnReceive ep r_opt pl")
+               apply (clarsimp simp: when_def split: option.splits)
+               apply (intro conjI impI allI; clarsimp simp: isReceive_def)
+                apply (corressimp corres: restart_thread_if_no_fault_corres[simplified isNothing_def])
+                apply (fastforce simp: obj_at_def)
+               apply (rule corres_guard_imp)
+                 apply (rule corres_split[OF _ reply_unlink_tcb_corres])
+                    apply (rule corres_guard_imp)
+                      apply (rule restart_thread_if_no_fault_corres[simplified isNothing_def])
+                     apply simp
+                    apply simp
+                   apply fastforce
+                  apply (wpsimp wp: reply_unlink_tcb_valid_sched_action)
+                 apply wpsimp
+                apply (fastforce simp: vs_all_heap_simps pred_tcb_at_def obj_at_def)
+               apply clarsimp
+              apply (prop_tac "\<not> isReceive st'")
+               apply (case_tac st; clarsimp simp: isReceive_def)
+              apply (case_tac st
+                     ; clarsimp simp: isReceive_def
+                     ; (corressimp corres: restart_thread_if_no_fault_corres[simplified isNothing_def]
+                        , fastforce simp: obj_at_def))
+             apply (wpsimp wp: gts_wp)
+            apply (wpsimp wp: gts_wp')
+           apply (clarsimp simp: vs_all_heap_simps obj_at_def is_tcb_def)
+          apply clarsimp
+         apply (intro hoare_vcg_conj_lift_left_fix; (solves \<open>wpsimp wp: gts_wp\<close>)?)
+           apply (wpsimp wp: restart_thread_if_no_fault_tcb_sts_of_other
+                             reply_unlink_tcb_tcb_sts_of_other gts_wp)
+         apply (rule hoare_weaken_pre
+               , rule cancel_all_ipc_loop_body_reply_unlink_ts_pred_other
```

Comment:
nitpick: interesting comma style. I was recently told that this is not what we usually do.

Reply:
Normally I place the comma one to the right. But I do prefer having the commas and the semi colons on the new line

Reply:
@Xaphiosis what's the style verdict on this one? I don't think we have commas on the new line anywhere yet. I don't like it for semicolons either, but I don't like any operators on the new line, so I'm just being consistent ;-)